### PR TITLE
Implement error.type attribute in HTTP semconv

### LINKED ISSUE #9438: Implement `error.type` attribute
See https://github.com/open-telemetry/semantic-conventions/pull/205

### PR DESCRIPTION
**Pull request by mateuszrzeszutek**
Resolves #9438
Part of #9090

### CODE REVIEW

File: instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/http/HttpCommonAttributesGetter.java [hunk 1]
```diff
@@ -53,4 +53,25 @@ public interface HttpCommonAttributesGetter<REQUEST, RESPONSE> {
...
+   * (if any was caught) or the value {@code _OTHER} will be used as error type.
+   *
+   * <p>The cardinality of the error type should be low. The instrumentations implementing this
+   * method are recommended to document the custom values they support.
+   *
+   * <p>Examples: {@code Bad Request}, {@code java.net.UnknownHostException}, {@code request
+   * cancelled}, {@code _OTHER}.
+   */
+  @Nullable
+  default String getErrorType(
```
Comment (mateuszrzeszutek) [inline instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/http/HttpCommonAttributesGetter.java:73]: I thought about creating a separate extractor just for that, but decided against it -- let's first see if there are any other use cases for this attribute, and we can retroactively implement a new getter interface if there's a need to do so.

File: instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/http/HttpStatusCodeConverter.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    serverPhrases.put(501, "Not Implemented");
+    serverPhrases.put(502, "Bad Gateway");
+    serverPhrases.put(503, "Service Unavailable");
+    serverPhrases.put(504, "Gateway Timeout");
+    serverPhrases.put(505, "HTTP Version Not Supported");
+    serverPhrases.put(506, "Variant Also Negotiates");
+    serverPhrases.put(507, "Insufficient Storage");
+    serverPhrases.put(508, "Loop Detected");
+    serverPhrases.put(510, "Not Extended");
+    serverPhrases.put(511, "Network Authentication Required");
```
Comment (trask) [inline instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/http/HttpStatusCodeConverter.java:72]: we changed this late in the PR to use the status code directly as the `error.type`, check out https://github.com/open-telemetry/semantic-conventions/pull/205#discussion_r1318747110

Comment (trask) [reply instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/http/HttpStatusCodeConverter.java:72]: I'll send a semconv PR to make this more explicit, I can see how this is unclear:

> SHOULD be set to the string representation of the status code

Comment (trask) [reply instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/http/HttpStatusCodeConverter.java:72]: https://github.com/open-telemetry/semantic-conventions/pull/331

Comment (mateuszrzeszutek) [reply instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/http/HttpStatusCodeConverter.java:72]: Huh, I was still under the impression we're supposed to use the reason phrase 😅 
I'll fix that, thanks for letting me know

**Review by laurit — APPROVED** [review]

**Review by trask — APPROVED** [review]
thx!

**Pull request merged**